src/v/coproc/event_listener.cc

```diff
@@ -252,4 +175,35 @@ event_listener::poll_topic(model::record_batch_reader::data_t& events) {
                                  : ss::stop_iteration::no;
 };
 
+ss::future<>
+event_listener::process_events(event_batch events, model::offset last_offset) {
+    for (auto& [type, prepared_events] : events) {
```

Comment:
why is `events` grouped by type? Why can't each event know what type it is?
```
for (auto & event : events) {
  for (auto* [type, hander] : _handlers) {
      if (event.type == type) {
        handler.process_event(event);
      }
  }
}
```

Reply:
Because `process` for `event_handler` gets batch with one `coproc_type`.
I do not want to run this method for each event from `events`.